### PR TITLE
fix(Header): Adjust header buttons spacing

### LINKED ISSUE #574: Header button left & right paddings are different
#### Environment
- __Package version(s)__: 0.53.2

#### Code Sandbox
Link to a minimal repro (fork [this code sandbox](https://codesandbox.io/s/itwinui-react-example-son74)): 
- Version with 0.38.0 version, where right & left paddings  for header button were same: https://codesandbox.io/s/itwinui-react-active-header-element-highlight-issue-1024k
- Version with 0.53.2 (latest for now) version, where right & left paddings for header button are different: https://codesandbox.io/s/itwinui-react-active-header-element-highlight-issue-forked-latest-packages-fyx6l7

#### Actual behavior
In current package version header button right & left button paddings are different.
image

#### Expected behavior
In previous package version those paddings were identical. 
image

### PR DESCRIPTION
**Pull request by veekeys**
I have changed left padding from 12 to 8 pixels. This saves horizontal space too. 
So just the look is up for debate.

image

UPDATE:
Padding left as was, just fixed the button. FOr this needed to add new class from dropdown button, so react will need an update.

Closes #574

### CODE REVIEW

File: backstop/tests/header.html [hunk 1]
```diff
@@ -172,6 +172,29 @@ <h2>Default</h2>
...
                 ></svg-caret-down-small>
               </button>
+
+              <svg-chevron-right
+                class="iui-chevron"
+                aria-hidden="true"
+              ></svg-chevron-right>
+
+              <!-- Version Button -->
+              <button
```
Comment (mayank99) [inline backstop/tests/header.html:182]: Why added this? Servers no purpose from what I can tell

Comment (veekeys) [reply backstop/tests/header.html:182]: There is no test on button without dropdown. Otherwise, we would have noticed this inconsistency.
So I added this to have all possible cases shown.
I could either leave this or replace iModel picker or some other button with this.

File: src/header/header.scss [hunk 2]
```diff
@@ -330,7 +330,7 @@
 
   > *:first-child {
     .iui-button {
-      padding-left: $iui-sm;
+      padding-left: $iui-s;
```
[the remark this answers is not in the document]
Comment (veekeys) [reply src/header/header.scss:333]: idk.. for me 12 px is random number and should be used on edge cases. Maybe this is the one, cant say.

File: backstop/tests/header.html [hunk 3]
```diff
@@ -172,6 +172,29 @@ <h2>Default</h2>
...
+                aria-label="Model picker"
+                aria-expanded="false"
+                aria-current="location"
+              >
+                <svg-model
+                  class="iui-button-icon iui-header-button-icon"
+                  aria-hidden="true"
+                ></svg-model>
+                <span class="iui-button-label">
+                  <div>Version Gama</div>
```
Comment (FlyersPh9) [inline backstop/tests/header.html:194]:
```suggestion
                  <div>Version Charlie</div>
```
You can use Charlie if you want to keep the NATO phonetic alphabet ordered correctly.

Comment (mayank99) [reply backstop/tests/header.html:194]: I thought this was the greek alphabet, hence gamma. http://stars.astro.illinois.edu/sow/greek.html

Just misspelled a little bit.
```suggestion
                  <div>Version Gamma</div>
```

File: src/header/header.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply src/header/header.scss:333]: > because the caret icon occupies invisible space.

He is correct, the icon takes up 16x16 but is visibly smaller than that.  In this screenshot I've added four 12px red blocks & one 16px green block to help demonstrate.
Screen Shot 2022-04-12 at 4 34 57 PM

[the remark this answers is not in the document]
Comment (FlyersPh9) [reply src/header/header.scss:333]: These changes do feel a little cramped but not so much that I'd dismiss the suggestion. If the reason for the change is so we can show a 3rd header button within a target device screen width, then I'd say that change makes sense.  But if this is just for more variable consistency, I'm more hesitant.

I think we could solve [the original issue](https://github.com/iTwin/iTwinUI/issues/574#issuecomment-1090477211) a different way.

**Comment by FlyersPh9** [comment]
If we do make this change it would require changing the styling within `menu.scss` to align correctly:
Screen Shot 2022-04-12 at 4 06 19 PM

**Review by elephantcatdog — APPROVED** [review]
Seems ok to me if the other comments are resolved.

File: src/header/header.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (veekeys) [reply src/header/header.scss:333]: Suggestions?

[the remark this answers is not in the document]
Comment (mayank99) [reply src/header/header.scss:333]: > I think normal padding should be 12px on both sides, and only split-button should have less on the right

^My suggestion

[the remark this answers is not in the document]
Comment (veekeys) [reply src/header/header.scss:333]: So when you say split-button, you also include dropdown button? Cause visually they look alike.

I still dont get the 12px part, though.
If I check our simple button, it has 16px padding, for example (on the left) and 8px on right.
image
Then if I check split button, left padding stays 16px, and 2nd button with arrow has only 4px right padding
image

[the remark this answers is not in the document]
Comment (veekeys) [reply src/header/header.scss:333]: Oh, should have added this button instead
image

With icon default one has 16px left padding.
Small one has 8px.
So again, 12px is some custom stuff, which I might just not understand the reasoning for :/

[the remark this answers is not in the document]
Comment (FlyersPh9) [reply src/header/header.scss:333]: I think the reasoning for 12px vs 16px was just visual preference.  I tend to lean towards Mayank's suggestion as well:  leave the padding at 12px unless a dropdown is present (split or not).

[the remark this answers is not in the document]
Comment (mayank99) [reply src/header/header.scss:333]: > So when you say split-button, you also include dropdown button?

Yeah I meant dropdown.

> If I check our simple button, it has 16px padding ...

> So again, 12px is some custom stuff, which I might just not understand the reasoning for

I agree 12px vs 16px is inconsistent. And it doesn't make sense to look at menu items because they are not aligned with our dropdown buttons. Maybe we need to find a common value that works for both default button and header buttons?
image

File: backstop/tests/header.html (hunk 3 above)
Comment (veekeys) [reply backstop/tests/header.html]: Yup, good catch. Did the fix.

File: src/header/header.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (veekeys) [reply src/header/header.scss]: Ok, I have created a new class for dropdown button and applied the needed padding.. Specificty needed to be higher to override header button, but.. Hopefully refactor will improve things there.
Now visuals hsould be as you wish, even though I am not a fan of those 🗡️ 

Also hpefully fixed menu alignment issue.

File: backstop/tests/menu.html [hunk 4]
```diff
@@ -1759,35 +1759,37 @@ <h3>Borderless button</h3>
       </ul>
     </section>
     <script>
-      const sectionPadding = parseInt(getComputedStyle(document.querySelector('#demo-header')).paddingLeft);
+      window.onload = () => {
```
Comment (mayank99) [inline backstop/tests/menu.html:1762]: This fixes flakiness with positioning calculations? I was trying to do something in #576 but couldn't figure it out

Comment (veekeys) [reply backstop/tests/menu.html:1762]: Check new images. This fixes the positioning of the menu.

File: src/header/header.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply src/header/header.scss]: > I have created a new class for dropdown button

Shouldn't this also apply to non-split dropdown buttons? So most of the images shouldn't change.

[the remark this answers is not in the document]
Comment (veekeys) [reply src/header/header.scss]: not sure what you have meant with your comment, but I did forget adding new class in `menu`, `skip-to-content` html files.
I did that and few images got rid of.
Now all images are changed either because of additional new button I added in demo, or menu position.

[the remark this answers is not in the document]
Comment (mayank99) [reply src/header/header.scss]: Yeah that makes sense. Ignore my comment 😅

**Review by mayank99 — APPROVED** [review]

**Review by FlyersPh9 — APPROVED** [review]
